types/recommendation.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2019 The MayaData Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CStorClusterRecommendation is a kubernetes custom
+// resource that represents recommended configurations
+// for one given input/requirement.
+type CStorClusterRecommendation struct {
```

Comment:
Why not CStorPoolClusterRecommendation?
@AmitKumarDas why do we want to avoid the term "Pool" when the actual resources are named like CStorPoolCluster and CStorPoolInstance?
Better to have a consistent naming pattern for all our resources.

Reply:
I would also prefer `CStorPoolClusterRecommendation`

Reply:
Its just the pattern of naming we were following till now.
The CSPC should ideally have been CStorCluster & this in turn should have created CStorPoolInstance & so on.
However, with DAOs, we thought to remove the term Pool & just stick with Cluster. The main motive was to avoid using Pools which are overused.

Hence we have CStorClusterConfig. Similarly, I suggested CStorClusterRecommendation.

The only demerit of above general naming will occur when we start designing DAOs for CStor Volumes. Whats the name we should use for that. We need to use CStorVolumeXYZ to avoid confusion.

cc @vishnuitta @kmova I would like to have a closure on this.